test/helpers/client_helper.rb

```diff
@@ -0,0 +1,35 @@
+##
+# Helper mixin for tests (especially compute-based ones). Provide access to the
+# client in use via the #client method to use.
+
+module ClientHelper
+  # Check to see if an operation is finished.
+  #
+  # @param op [Excon::Response] the operation object returned from an api call
+  # @return [Boolean] true if the operation is no longer executing, false
+  #   otherwise
+  def operation_finished?(op)
```

Comment:
I almost used the operation model, but I felt it was confusing because once you wrap a returned result with it, you no longer have access to the underlying excon result. To me, it was important to be able to assert low-level request properties on the request (like status code). It also tends to make the tests easier to understand (e.g. imagine tests on methods that return plain excon results mixed with tests that return operation model objects and assert on that).

If I weren't writing request-level tests though, I'd definitely be using the operation model here.